oraclebmc/util.py

```diff
@@ -99,3 +100,14 @@ def __bool__(self):
         return self.truthy
     # PY2 Compatibility
     __nonzero__ = __bool__
+
+def initkwargs(fn):
```

Comment:
This is a nice solution!

I think we can expand the utility of this pattern a bit by moving the decorator to the class instead of the `__init__` method.  Then, we'll have a single place to inject additional post-processing in the future.

```python
@util.apply_enhancements  # or some name to indicate generality
class SomeModel(object):
    def __init__(...)
```

[The PEP for class decorators](https://www.python.org/dev/peps/pep-3129/) suggests this is 3.0+ only, but things work fine here with 2.7.12.

Testing also gets easier when the individual enhancements can be defined on their own, with the actual decorator simply applying each function.  Since the decorator returns the original class, other decorators can be stacked above this one:

```python
# model.py
@user_defined_decorator
@util.extend
class SomeModel(object):
    ...


# util.py
def kwarg_init(self, **kwargs):
    # TODO inspect via swagger_types
    pass


def extend_model(cls):
    cls.__init__ = kwargs_init
    return cls
```